### PR TITLE
Closes #1321 #1220 #1306 #1219 doc updates

### LINKED ISSUE #1321: Documentation: Display CRAN Date on Changelog Page
### Please select a category the issue is focused on?

Other

### Let us know where something needs a refresh or put your idea here!

The CRAN release date should be displayed on the Changelog page. (Update `_pkgdown.yml`:
```
news:
  cran_dates: true
```
)

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
Hi all,  

I did a few teaks to our documentation for this PR.  

*  Update on release schedule for readme 
*  Update for release strategy and schedule for github vignette
*  Update on what each workflow does in PR vignette
*   Quick pkgdown update for cran release date

I also started a hotfix checklist in the release checklist, which is just a sub-heading at the moment.  @thomas-neitmann  do you mind putting the hotfix steps into the comments here and I will update the `release.md` file.  I'm not sure where this checklist should live, but it is nice to have when we do a release for all the reminders.  Originally, it was supposed to pre-populate with a different PR checklist of devel into main,  but did not work...

**Comment by rossfarrugia** [comment]
@bms63 will take a look later this week, thanks for the updates! did you mean this to merge with `main` though? think it should be `devel`

**Comment by bms63** [comment]
@rossfarrugia fixed! newbie mistake

**Comment by bms63** [comment]
Hi all,

@pharmaverse/admiral I have made updates to the Github Vignette on the Release Strategy if you are curious about how we are doing a release.  @thomas-neitmann, @bundfussr does this capture our conversation at the meeting today?

image

File: vignettes/git_usage.Rmd
Comment (bundfussr) [inline vignettes/git_usage.Rmd:184]: I think we can remove this step. As we use `pre-release` for the final checks and updates, no freeze of `devel` is required.

File: vignettes/git_usage.Rmd
Comment (bundfussr) [inline vignettes/git_usage.Rmd:186]: "into" is missing before "the `main` branch".

File: vignettes/git_usage.Rmd
Comment (bundfussr) [inline vignettes/git_usage.Rmd:195]: Should we protect these branches and do not delete them after merging?

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: vignettes/git_usage.Rmd
Comment (rossfarrugia) [reply vignettes/git_usage.Rmd:185]: also do we need to link CRAN?

File: vignettes/git_usage.Rmd
Comment (rossfarrugia) [inline vignettes/git_usage.Rmd:204]: why repeat this section from the README? you'll have to update it in 2 places then. i would just leave it on the README only

File: vignettes/pr_review_guidance.Rmd
Comment (rossfarrugia) [inline vignettes/pr_review_guidance.Rmd:96]: Reminder later to add here where the source workflows are housed, e.g. {admiraldev} or {admiralci} once we decide

File: vignettes/pr_review_guidance.Rmd
Comment (rossfarrugia) [inline vignettes/pr_review_guidance.Rmd:100]: boilderplate -> boilerplate

File: vignettes/pr_review_guidance.Rmd
Comment (rossfarrugia) [inline vignettes/pr_review_guidance.Rmd:104]: "checks  the" - double space

File: vignettes/pr_review_guidance.Rmd
Comment (rossfarrugia) [inline vignettes/pr_review_guidance.Rmd:136]: is `[][multi-version-docs]` correct - not seen a blank [] before so not sure how this renders

**Review by rossfarrugia — CHANGES_REQUESTED** [review]

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd:195]: I was thinking about that...

For the patch branch, if you patch infrequently then the patch branch might get out of synch and sometime spent figuring out what to merge in.  Better to delete and re-branch off the current main for patches.

For pre-release - this might get tricky again.  We might have a pre-release for patch/hotfix and a pre-release for a quarterly release.  But I guess this should just be a mirror of the main anyways.  

So...maybe pre-release is kept and protected while patches are deleted?

File: README.md
Comment (bms63) [reply README.md:49]: sure np - it was just placeholder :)

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd:185]: no i will remove

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd:204]: I kind of like it being here.  I will be make sure its gets updated - its on the release checklist when updating the description and changelog files - so i don't forget

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd:195]: Perhaps there is a way to always keep the main and pre-release branch in sync?

Comment (bms63) [reply vignettes/git_usage.Rmd:195]: makes sense.  there is a github action that can help automate this so we don't lose track of it.

https://github.com/marketplace/actions/sync-branches @dinakar29.  Is this a good one to keep our main and patch branch in sync?

Comment (dinakar29) [reply vignettes/git_usage.Rmd:195]: Looks pretty good to me 👍🏽

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd:184]: I'm thinking if we need to do some emergency fixes before the release having other stuff going into `devel` besides our emergency fixes might get a bit muddled.  If you we merge our fixes into pre-release then we have to remember to put them back into devel as well, which can get tedious. 

I think the freezes are nice as then all of the core developers can really focus on release activities.

Comment (bundfussr) [reply vignettes/git_usage.Rmd:184]: I would prefer doing the emergency fixes in `pre-release` and merging `pre-release` back into `devel` after the release. Then we do not need to rely on that everyone has seen the freeze message. It is also requires less PRs. Because if we do the emergency fixes in `devel`, we need two PRs for each fix, one for merging the feature branch into `devel` and another one for merging `devel` into `pre-release`.

Comment (bms63) [reply vignettes/git_usage.Rmd:184]: that is a good point.  i was thinking the PRs were getting a bit excessive

File: vignettes/pr_review_guidance.Rmd
Comment (bms63) [reply vignettes/pr_review_guidance.Rmd:104]: dang you got good eyes

File: vignettes/pr_review_guidance.Rmd
Comment (bms63) [reply vignettes/pr_review_guidance.Rmd:136]: So I just straight copied this from admiraltemplate without reading it too thoroughly.  I have now removed all the brackets/links in the workflows as they are not set up the same here.

File: vignettes/git_usage.Rmd
Comment (bms63) [reply vignettes/git_usage.Rmd]: Created an issue #1332

File: vignettes/git_usage.Rmd
Comment (rossfarrugia) [reply vignettes/git_usage.Rmd:207]: @bms63 couldn't it just be linked back to the main page from the README via a link here? sorry but i have a real bugbear about exactly the same info being maintained in 2 places. @thomas-neitmann any thoughts??

Comment (bms63) [reply vignettes/git_usage.Rmd:207]: i can just do a link.  np at all

Comment (bms63) [reply vignettes/git_usage.Rmd:207]: link provided, but it won't go to the updated readme until the website is re-depolyed.  I couldn't figure out where the `.html` exists for the readme in the sitemap..  

@dinakar29 do you know how I can link to the updated readme on the pkgdown site when I create a local version for the release schedule.

We can link do different pages and sections using e.g. (programming_strategy.html#deprecation-1), but this doesn't works for (readme.html#release-schedule).

Comment (bundfussr) [reply vignettes/git_usage.Rmd:207]: Maybe `../index.html#release-schedule`?

Comment (bms63) [reply vignettes/git_usage.Rmd:207]: worked like a charm!

File: vignettes/git_usage.Rmd
Comment (dinakar29) [reply vignettes/git_usage.Rmd]: Cool, I'll look into this 👆

**Review by rossfarrugia — APPROVED** [review]

**Review by bundfussr — APPROVED** [review]

**Pull request merged**
